Coredumped with initial sync at 35 block

I[2020-05-02|19:32:39.794] Committed state                              module=state height=34 txs=0 appHash=7EAA7AA06A358B5B52C202852CDB3EE400EA29E63C12B75130AFAC257D9AD426
I[2020-05-02|19:32:39.810] Executed block                               module=state height=35 validTxs=0 invalidTxs=0
I[2020-05-02|19:32:39.834] Committed state                              module=state height=35 txs=0 appHash=DECC2CD7733EA91371BB2C991A693A96FB0F6A10E72F393C058ED49C29362983
SwingSet: controller: UnhandledPromiseRejectionWarning: TypeError <Object <Object <[Object: null prototype] {}>>>: Cannot read property 'receive' of null
    at eval (/bundled-source/eventual-send.cjs.js:535:23)
    at Object.eval [as applyMethod] (/bundled-source/eventual-send.cjs.js:522:18)
    at eval (/bundled-source/eventual-send.cjs.js:571:66)
SwingSet: controller: UnhandledPromiseRejectionWarning: TypeError <Object <Object <[Object: null prototype] {}>>>: Cannot read property 'receive' of null
    at eval (/bundled-source/eventual-send.cjs.js:535:23)
    at Object.eval [as applyMethod] (/bundled-source/eventual-send.cjs.js:522:18)
    at eval (/bundled-source/eventual-send.cjs.js:571:66)
SwingSet: controller: UnhandledPromiseRejectionWarning: TypeError <Object <Object <[Object: null prototype] {}>>>: Cannot read property 'receive' of null
    at eval (/bundled-source/eventual-send.cjs.js:535:23)
    at Object.eval [as applyMethod] (/bundled-source/eventual-send.cjs.js:522:18)
    at eval (/bundled-source/eventual-send.cjs.js:571:66)
SwingSet: controller: UnhandledPromiseRejectionWarning: TypeError <Object <Object <[Object: null prototype] {}>>>: Cannot read property 'receive' of null
    at eval (/bundled-source/eventual-send.cjs.js:535:23)
    at Object.eval [as applyMethod] (/bundled-source/eventual-send.cjs.js:522:18)
    at eval (/bundled-source/eventual-send.cjs.js:571:66)
SwingSet: controller: UnhandledPromiseRejectionWarning: TypeError <Object <Object <[Object: null prototype] {}>>>: Cannot read property 'receive' of null
    at eval (/bundled-source/eventual-send.cjs.js:535:23)
    at Object.eval [as applyMethod] (/bundled-source/eventual-send.cjs.js:522:18)
    at eval (/bundled-source/eventual-send.cjs.js:571:66)
SwingSet: controller: UnhandledPromiseRejectionWarning: TypeError <Object <Object <[Object: null prototype] {}>>>: Cannot read property 'receive' of null
    at eval (/bundled-source/eventual-send.cjs.js:535:23)
    at Object.eval [as applyMethod] (/bundled-source/eventual-send.cjs.js:522:18)
    at eval (/bundled-source/eventual-send.cjs.js:571:66)
SwingSet: controller: UnhandledPromiseRejectionWarning: TypeError <Object <Object <[Object: null prototype] {}>>>: Cannot read property 'receive' of null
    at eval (/bundled-source/eventual-send.cjs.js:535:23)
    at Object.eval [as applyMethod] (/bundled-source/eventual-send.cjs.js:522:18)
    at eval (/bundled-source/eventual-send.cjs.js:571:66)
SwingSet: controller: UnhandledPromiseRejectionWarning: TypeError <Object <Object <[Object: null prototype] {}>>>: Cannot read property 'receive' of null
    at eval (/bundled-source/eventual-send.cjs.js:535:23)
    at Object.eval [as applyMethod] (/bundled-source/eventual-send.cjs.js:522:18)
    at eval (/bundled-source/eventual-send.cjs.js:571:66)
SwingSet: controller: UnhandledPromiseRejectionWarning: TypeError <Object <Object <[Object: null prototype] {}>>>: Cannot read property 'receive' of null
    at eval (/bundled-source/eventual-send.cjs.js:535:23)
    at Object.eval [as applyMethod] (/bundled-source/eventual-send.cjs.js:522:18)
    at eval (/bundled-source/eventual-send.cjs.js:571:66)
SwingSet: controller: UnhandledPromiseRejectionWarning: TypeError <Object <Object <[Object: null prototype] {}>>>: Cannot read property 'receive' of null
    at eval (/bundled-source/eventual-send.cjs.js:535:23)
    at Object.eval [as applyMethod] (/bundled-source/eventual-send.cjs.js:522:18)
    at eval (/bundled-source/eventual-send.cjs.js:571:66)
I[2020-05-02|19:32:39.900] Executed block                               module=state height=36 validTxs=1 invalidTxs=0
I[2020-05-02|19:32:39.909] Committed state                              module=state height=36 txs=1 appHash=98183558ED6B1435555DDB1F0DB1F9EE5407B7B555BE4EBB10C85BF57B6CD5F7
panic: Failed to process committed block (37:A28238E0E478CAF76A68FE4ECF33409295D9338936FD711C2A519F193107073A): wrong Block.Header.AppHash.  Expected 98183558ED6B1435555DDB1F0DB1F9EE5407B7B555BE4EBB10C85BF57B6CD5F7, got E5A7223C0B605358E7B770D4ADBC28E6EB5A9348E143206E6DE4FBAFB765ED53

goroutine 91 [running]:
github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).poolRoutine(0xc00010d1e0)
        /root/go/pkg/mod/github.com/tendermint/tendermint@v0.33.4/blockchain/v0/reactor.go:349 +0x12ef
created by github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).OnStart
        /root/go/pkg/mod/github.com/tendermint/tendermint@v0.33.4/blockchain/v0/reactor.go:118 +0x86
Aborted (core dumped)

--- a/packages/cosmic-swingset/lib/chain-main.js
+++ b/packages/cosmic-swingset/lib/chain-main.js
@@ -66,7 +66,7 @@ export default async function main(progname, args, { path, env, agcc }) {
       },
       rej => {
         // console.error(`Rejecting in Node to ${str} with`, rej);
-        replier.reject(`rejection ignored: ${rej.stack || rej}`);
+        replier.reject(`${(rej && rej.stack) || rej}`);
       },
     );
   }
@@ -200,6 +200,9 @@ export default async function main(progname, args, { path, env, agcc }) {
     // console.log(`toSwingSet`, action, replier);
     if (action.type === AG_COSMOS_INIT) {
       // console.error('got AG_COSMOS_INIT', action);
+      if (!process.env.BOOT_ADDRESS) {
+        throw Error(`You must set $BOOT_ADDRESS to run a chain node`);
+      }
       portNums.dibc = action.ibcPort;
       return true;
     }

